sshd-core/src/main/java/org/apache/sshd/agent/common/AbstractAgentProxy.java

```diff
@@ -99,24 +117,44 @@ public void setShutdownOnExit(boolean shutdown) {
 
     @Override
     public byte[] sign(PublicKey key, byte[] data) throws IOException {
-        Buffer buffer = createBuffer(SshAgentConstants.SSH2_AGENTC_SIGN_REQUEST);
+        byte cmd = SshAgentConstants.SSH2_AGENTC_SIGN_REQUEST;
+        byte okcmd = SshAgentConstants.SSH2_AGENT_SIGN_RESPONSE;
+        if (FactoryManager.AGENT_FORWARDING_TYPE_IETF.equals(channelType)) {
+            cmd = SshAgentConstants.SSH_AGENT_PRIVATE_KEY_OP;
+            okcmd = SshAgentConstants.SSH_AGENT_OPERATION_COMPLETE;
+        }
+
+        Buffer buffer = createBuffer(cmd);
+        if (FactoryManager.AGENT_FORWARDING_TYPE_IETF.equals(channelType)) {
+            buffer.putString("sign");
+        }
         buffer.putPublicKey(key);
```

Comment:
The code does not seem equivalent to previous one - there is now a **new** created buffer.